hangman/lib/hangman.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Hangman do
+  @moduledoc """
+  Documentation for Hangman.
+  Christian Ventura
+  """
+
+  #the defined struct
+  defstruct game_state: :initializing, turns_left: 7, letters: [], used: [], last_guess: "", word: ""
+
+
```

Comment:
As you're writing code, I'd like you to step back every now and then and think about what you're writing. Look at it with a critical eye, and ask yourself questions such as "should this be this hard?".

For example, the code for `word_to_blank` is really complex. But what it it actually doing? Well, we could specify it as
"take a word and a list of letters that have already been guessed. Create a new word where letters in the original that haven't been guesses yet are replaced by underscores".

We're mapping one list (the original word) to another (the word with blanks), so `Enum.map` seems like a good place to start. Then we just need a function that takes a letter from the word and the list of guesses and returns either that letter or underscore (just as the spec says). Maybe something like:

~~~ elixir
 def word_to_blank(word, used) do
    word
    |> Enum.map(&underscore_unless_used(&1, used))
  end

  def underscore_unless_used(letter, used) do
    if letter in used do
      letter
    else
      "_"
    end
  end
~~~